### PR TITLE
Python: Promote modeling of SQLAlchemy

### PR DESCRIPTION
**Pull request by RasmusWL**
After creating elaborate tests for how you can use the ORM to construct queries, I realized it would be difficult to make our normal SQL injection query ever flag those, so did a 180 on how to handle it, and ended up adding a new query.

We could also extend our existing SQL-injection query to _also_ flag up TextClause injection. I felt that putting them together would end up being a bit confusing, but I'm very open to changing this if it is cause for concern.

### CODE REVIEW

File: python/ql/src/Security/CWE-089/SQLAlchemyTextClauseInjection.qhelp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+<p>
+In the second case, the query is built using ORM models, but part of it is using a
+<code>TextClause</code> directly including a user-supplied input. Since the
+<code>TextClause</code> is inserted directly into the final SQL, this code is vulnerable
+to a SQL injection attack.
+</p>
+
+<p>
+In the third case, the query is built fully using the ORM models, so in the end, the
+user-supplied input will passed passed to the database using query parameters. The
```
Comment (yoff) [inline python/ql/src/Security/CWE-089/SQLAlchemyTextClauseInjection.qhelp:43]:
```suggestion
user-supplied input will be passed to the database using query parameters. The
```

File: python/ql/test/query-tests/Security/CWE-089-SQLAlchemyTextClauseInjection/test.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    username_formatted_for_sql = sqlalchemy.text("'{}'".format(username))
+    stmt = sqlalchemy.select(User).where(User.username == username_formatted_for_sql)
+    results = session.execute(stmt).scalars().all()
+
+    # GOOD, does not allow for SQL injection
+    stmt = sqlalchemy.select(User).where(User.username == username)
+    results = session.execute(stmt).scalars().all()
+
+
+    # All of these should be flagged by query
```
Comment (yoff) [inline python/ql/test/query-tests/Security/CWE-089-SQLAlchemyTextClauseInjection/test.py:40]: ...and they are :-) I guess that would be more clear from an inline test.

**Review by yoff — CHANGES_REQUESTED** [review]
Quite impressive testing here and also the work to complete the modelling 💪 
I thought it over, and it does seem quite reasonable (as in not overly FP-prone) to always alert on text objects being user controlled. If such objects are not executed, the code can probably be simplified and the alert would have been still useful. So the two-query-approach seems fine here.

It might be nice to add an inline style test for the bonus query, but I am happy to merge without it. The requested changes only really comprise the one typo I spotted...

**Comment by RasmusWL** [comment]
> It might be nice to add an inline style test for the bonus query, but I am happy to merge without it. The requested changes only really comprise the one typo I spotted...

Yes. I think we have something to learn from JS here, since they have a standard test setup where they use something like `# $ OK` and `# $ NOT_OK` to ensure proper flow.

**Review by yoff — APPROVED** [review]
LGTM as soon as the subpaths are added to the test expectations.

**Review by yoff — APPROVED** [review]
LGTM

**Pull request merged**